### PR TITLE
Skippable vSphere user validations

### PR DESCRIPTION
**Pull request by pokearu**
*Issue #5865:*

*Description of changes:*
1. Modified to fail on vsphere missing user permissions.
2. `--skip-validations` flag was initally only done for upgrade preflights. Refactored the code to have skip validations for create cluster and also pass it to provider level validations.

*Testing:*
Created a user with missing permissions and tested validations. Now the error shows as

```bash
2023-07-14T20:09:04.809Z  V3  user ramaliar-test@vsphere.local missing vSphere permissions  {"Permissions": "- username: ramaliar-test@vsphere.local\n  objectType: Network\n  path: /Denver/network/Dev\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: Datastore\n  path: /Denver/datastore/datastore1.1\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: ResourcePool\n  path: /Denver/host/EKS-A/Resources/CI\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: Folder\n  path: /Denver/vm/Dev/capv/ramaliar\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: VirtualMachine\n  path: /Denver/vm/Templates/bottlerocket-v1.25.11-kubernetes-1-25-eks-17-amd64-327823e\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n- username: ramaliar-test@vsphere.local\n  objectType: Folder\n  path: /Denver/vm/Templates\n  permissions:\n  - VirtualMachine.Interact.PowerOff\n  - VirtualMachine.Interact.PowerOn\n"}
2023-07-14T20:09:04.810Z  V0  ❌ Validation failed {"validation": "vsphere Provider setup is valid", "error": "validating vsphere user privileges: user ramaliar-test@vsphere.local missing vSphere permissions", "remediation": ""}
```

If we pass --skip-validations, the check is skipped.
```bash
root@ip-10-0-59-81:/home/ubuntu# ./eksctl-anywhere create cluster --help
This command is used to create workload clusters

Usage:
  anywhere create cluster -f <cluster-config-file> [flags]

Flags:
...
      --skip-validations stringArray        Bypass create validations by name. Valid arguments you can pass are --skip-validations=vsphere-user-privilege
...
```

Tested creating a cluster by skipping the check
```bash
./eksctl-anywhere create cluster -f ./config.yaml --skip-validations=vsphere-user-privilege
```

*Documentation added/planned :*
Plan to document the new flag and change in user experience.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/validations/createvalidations/createvalidations.go [hunk 1]
```diff
@@ -4,6 +4,16 @@ import (
 	"github.com/aws/eks-anywhere/pkg/validations"
 )
 
+// string values of supported validation names that can be skipped.
+const (
+	VSphereUserPriv = "vsphere-user-privilege"
```
Comment (jonathanmeier5) [inline pkg/validations/createvalidations/createvalidations.go:9]: We're going to document a list of these right? is there any way we can automate publishing what these are?

**Review by jonathanmeier5 — COMMENTED** [review]
There's a lot going on here. Would it be possible to break this PR up as follows?

- PR with code/CLI arguments necessary to skip validations
- follow up PRs for each distinct skip validation

If there's no way to break up the PR we can review and merge as-is, but it would be better to break this up if possible.

File: pkg/validations/createvalidations/createvalidations.go (hunk 1 above)
Comment (pokearu) [reply pkg/validations/createvalidations/createvalidations.go:9]: Publishing on the docs you mean? 
I dont think so, new flags we add will get listed with `--help` command but not sure if we can send that to the docs.

**Comment by pokearu** [comment]
> There's a lot going on here. Would it be possible to break this PR up as follows?
> 
> * PR with code/CLI arguments necessary to skip validations
> * follow up PRs for each distinct skip validation
> 
> If there's no way to break up the PR we can review and merge as-is, but it would be better to break this up if possible.

So this PR adds only one skip, the rest is just moving code around to make the skip more generic (originally it was only for upgrade pre-flights).
I could break the PR but it would still have all the files except the [validator.go](https://github.com/aws/eks-anywhere/pull/6182/files#diff-238036d622b0e49427d5c5115ba7fa29145e3741beb69f67aa345a6094280dee).

**Comment by pokearu** [comment]
Final PRs
https://github.com/aws/eks-anywhere/pull/6190
https://github.com/aws/eks-anywhere/pull/6193

**Pull request closed without merging**